### PR TITLE
Add keeper registry 1.2 contract and interface

### PR DESCRIPTION
**Pull request by infiloop2**
In this PR I'm adding the new keeper registry contract (v1.2) and also supporting the old one
Did a bit of refactoring to pass the parameters. The interface for the contract remains same, but some functions have different implementation depending on the registry version

Corresponding CL repo PR: CL repo PR: https://github.com/smartcontractkit/chainlink/pull/6646
where both registry versions are tested in smoke tests

**Note**: I ran `make compile_contracts` and it also updated the other wrappers with latest abigen

### CODE REVIEW

File: contracts/contract_models.go [hunk 1]
```diff
@@ -186,123 +186,6 @@ type MockGasFeed interface {
...
-		upkeepAddr string,
-		gasLimit uint32,
-		adminAddr string,
-		checkData []byte,
-		amount *big.Int,
-		source uint8,
-	) ([]byte, error)
-	Fund(ethAmount *big.Float) error
-}
-
```
Comment (infiloop2) [inline contracts/contract_models.go:210]: consolidated these all in contracts/ethereum_keeper_contracts.go

File: contracts/contract_models.go [hunk 2]
```diff
@@ -186,123 +186,6 @@ type MockGasFeed interface {
 	Address() string
 }
 
-type UpkeepRegistrar interface {
```
Comment (infiloop2) [inline contracts/contract_models.go:189]: consolidated these all in contracts/ethereum_keeper_contracts.go

File: .gitignore [hunk 3]
```diff
@@ -27,8 +27,8 @@ artifacts/
 # JUnit Test Reports
 junit.xml
 tests-*.xml
-logs/
-bin/
+/logs/
+/bin/
```
Comment (infiloop2) [inline .gitignore:31]: Before this was ignoring the bin generated for contracts, but I saw that there are already files committed for those so clarified this for consistency

**Review by tateexon — APPROVED** [review]
looks good to me

**Pull request merged**